### DISCUSSION TITLE
Is there a separate Olive/ONNX optimization script?

### DISCUSSION THREAD

**Discussion by R-N** [lshqqytiger/stable-diffusion-webui-directml#353] [Q&A] [answered]
Olive/ONNX optimization requires a lot of RAM and VRAM. My device can run the WebUI just fine with directml, but can't optimize checkpoints due to insufficient memory. So I want to use Google Colab just to optimize the checkpoint, but Colab prohibits WebUI and automatically terminates it. Is there a separate script for the optimization, perhaps? So I can use it in Colab.

By the way, I was assuming running the optimized checkpoint with Olive doesn't require significantly more RAM than normal checkpoint with just directml. Or does it?

Thank you

**Comment by lshqqytiger** [comment, marked as answer]
You have two options.
1. Download optimized model from huggingface instead of optimizing models yourself.
2. Try [olive branch of vladmandic/automatic](https://github.com/vladmandic/automatic/wiki/ONNX-Olive#olive-experimental-support) which requires less RAM.

**Reply by R-N** [reply]
I can't find olive-optimized version of the model I want to use.

I tried ripping the optimization script and running it on Colab. It didn't work as it needed DmlExecutionProvider and onnxruntime-directml. DirectML simply doesn't support linux because no directx. 

Thanks for the alternative, I'll try it when I have the time.
